src/main/java/org/apache/commons/lang3/reflect/TypeUtils.java

```diff
@@ -424,6 +424,11 @@ private static boolean isAssignable(final Type type, final ParameterizedType toP
             return true;
         }
 
+        //the component type should not be assignable to the array type
```

Comment:
This comment is backward.

I've committed a variation of the PR to git master with more tests.

I am wondering about other Type extensions. What do you think about `ypeUtilsTest.testIsAssignableGenericArrayTypeToWildercardType()`? See `isAssignable(Type, Type, Map<TypeVariable<?>, Type>)`.